react-app/src/library/services/vendor-implementations/jabra/jabra.ts

```diff
@@ -0,0 +1,322 @@
+import { VendorImplementation, ImplementationConfig } from "../vendor-implementation";
+import DeviceInfo from "../../../types/device-info";
+import { timedPromise } from "../../../utils";
+import { v4 } from 'uuid';
+import {
+    IApi,
+    EasyCallControlFactory,
+    CallControlFactory,
+    IMultiCallControl,
+    SignalType,
+    ICallControl,
+    ErrorType
+} from '@gnaudio/jabra-js';
+
+const incomingMessageName = 'jabra-headset-extension-from-content-script';
+const outgoingMessageName = 'jabra-headset-extension-from-content-script';
+const clientId = v4();
+const jabraVendorId = '0x0b0e';
+
+export default class JabraService extends VendorImplementation {
+    private static instance: JabraService;
+    static connectTimeout = 5000;
+
+    isConnecting = false;
+    isActive = false;
+    devices: Map<string, DeviceInfo>;
+    activeDeviceId: string = null;
+    isMuted = false;
+    isHeld = false;
+    version: string = null;
+    _connectDeferred: any; // { resolve: Function, reject: Function }
+    device = null;
+    jabraSdk: IApi;
+    easyCallControlFactory: EasyCallControlFactory;
+    callControlFactory: CallControlFactory;
+    multiCallControl: IMultiCallControl;
+    callControl: ICallControl;
+    ongoingCalls: number = 0;
+    callLock: boolean = false;
+
+    private constructor(config: ImplementationConfig) {
+        super(config);
+        this.vendorName = 'Jabra';
+        this.devices = new Map<string, DeviceInfo>();
+    }
+
+    deviceLabelMatchesVendor(label: string): boolean {
+        return label.toLowerCase().includes('jabra');
+    }
+
+    static getInstance(config: ImplementationConfig) {
+        if (!JabraService.instance) {
+            JabraService.instance = new JabraService(config);
+        }
+
+        return JabraService.instance;
+    }
+
+    get deviceInfo(): DeviceInfo {
+        if (this.activeDeviceId === null || this.devices.size === 0) {
+            return null;
+        }
+
+        return this.devices.get(this.activeDeviceId);
+    }
+
+    get deviceName(): string {
+        return this.deviceInfo.deviceName;
+    }
+
+    get isDeviceAttached(): boolean {
+        return !!this.deviceInfo;
+    }
+
+    resetState(): void {
+        this.setHold(null, false);
+        this.setMute(false);
+    }
+
+    async _processEvents(callControl: ICallControl) {
+        callControl.deviceSignals.subscribe(async (signal) => {
```

Comment:
1.) I suppose it would technically...
2.) Perhaps while the call is going? I have noticed that not during a call, I don't see anything reacting on the headset remote.
3.) I don't see why not.  We could maybe unsubscribe on `disconnect()`?

Honestly I don't have a 100% concrete answer for these but these are good points.